clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -353,14 +353,16 @@ else if (strategy == OffsetResetStrategy.LATEST)
             throw new NoOffsetForPartitionException(partition);
 
         log.debug("Resetting offset for partition {} to {} offset.", partition, strategy.name().toLowerCase(Locale.ROOT));
-        long offset = getOffsetsByTimes(Collections.singletonMap(partition, timestamp)).get(partition).offset();
+        long offset = getOffsetsByTimes(Collections.singletonMap(partition, timestamp), Long.MAX_VALUE).get(partition).offset();
```

Comment:
We can do this in a separate JIRA, but there is a TODO in this class to send the list offsets requests in parallel when resetting multiple partitions (currently we reset each partition separately, which means one ListOffset request for each partition). Should be an easy refactor after the changes from this KIP. I'll open the JIRA.